### ISSUE TITLE
Eliminate "all" for single chromosome assemblies

### ISSUE THREAD

**Issue by jrobinso** [aidenlab/juicebox.js#236]
In the special case of single chromosome assemblies, there is no difference in  "All" and the chromosome.     So for this case make 2 changes

(1) add the whole-genome resolution to the list of explicitly specified ones
(2) eliminate "All" from the chromosome pulldown menu

**Comment by lidaof** [comment]
Hi @jrobinso , can I used 'all' in the 3rd parameter for straw API?
`straw.getContactRecords(normalization,region1, 'all', units, binSize)`?

tried, get a error with `cannot getLength`

**Comment by jrobinso** [comment]
That is not correct syntax.  The region argument is an object with chr, start, and end.

**Comment by jrobinso** [comment]
The "all" query would not work even with correct syntax.   I just pushed a fix for that,  but the "region" arguments should look like this, do not specify start and end.

```
{
   chr: "ALL"
}
```
Also, do not specify a binSize, it is not relevant for "ALL".   To use this you will need to pull and build (with grunt).

In the future please open a new issue for a new problem or question,  this is not related to the original issue.  It helps with tracking.

**Comment by jrobinso** [comment]
BTW the "all" chromosome is used only for display, it is not really useful or suitable for analysis.   The units are kb (not bp), and original chromosome information is lost.

**Comment by lidaof** [comment]
Thanks @jrobinso . I was thinking it's kinda related so didn't open a new issue.
From what you said, suppose I should not use 'all' to get all the interactions (in genome) with one region?

Currently what I am doing is put all chrom object (with chr, start, end) in an array and loop it with my single query region to get all interactions. That;s why I was asking if I can use all instead of a for loop.

Thanks again.

**Comment by jrobinso** [comment]
It depends on what you are doing.  If you are doing analysis I would not use "ALL",  its is just used to get an approximate coarse scale view for visualization.   You can't even know the bin size easily.   If you want to paint a whole genome summary like juicebox does it is fine,  but remember the coordinates are kilo-bases from the start of the first chromosome.

**Comment by lidaof** [comment]
@jrobinso Thank you so much for clarifying.

**Comment by turner** [comment]
> *This was generated by AI during triage.*

## Triage Notes

Kept open — this is the one survivor of the 2017–2018 sweep. Labelled `enhancement` + `needs-triage`.

**What we've established so far:**

- **Still unimplemented.** `ChromosomeSelector.respondToDataLoadWithDataset` (`js/chromosomeSelector.js:69`) maps *every* entry of `dataset.chromosomes` into the pulldown, index 0 — "All" — included. There is no single-chromosome branch anywhere on that path.
- **Part (1) is partly there.** `HiCResolutionSelector` already special-cases whole-genome by substituting `browser.dataset.wholeGenomeResolution` and relabelling the axis mb-vs-kb (`js/hicResolutionSelector.js:84`, `:118`). What is missing is exposing that resolution in the ordinary list for a single-chromosome assembly, which is what this issue asked for.
- **`genome.js` already knows about the "all" pseudo-chromosome**, filtering it by name at `:49` and `:171` — so there is prior art for the test, and the question is which layer should apply it.
- **This is still live.** The open #398 ("Issues with HiC map scale for large (~2.4 Gbp) single scaffold genomes") is the same territory: single-scaffold assemblies where "All" and the sole chromosome are the same view. These two should be evaluated together — a fix here may be part of that one.
- The off-topic straw API sub-thread from 2018 (@lidaof) was answered at the time and is unrelated to the request.

**What we still need before this can be specified:**

- Should the filter be *"the assembly has one chromosome besides `all`"*, or the broader *"`all` and the sole chromosome describe the same view"*? The second also covers assemblies where a single scaffold dominates.
- Does dropping "All" from the pulldown need a matching guard on the state/session side, so an old session or a pasted URL naming `chr 0` still restores rather than throwing?
- Is this separable from #398, or is it a sub-task of it?

Sibling issues #95, #126, #177, #220, #226, #265, #266, #267 and #273 were closed in the same pass.

**Comment by turner** [comment]
> *This was generated by AI during triage.*

Moved `needs-triage` → `ready-for-human`.

**Why it can't be delegated:** the questions in the triage notes above are scope decisions, not lookups. Whether the filter is *"one chromosome besides `all`"* or the broader *"`all` and the sole chromosome describe the same view"* changes what gets built, and whether this is separable from #398 or a sub-task of it is a judgment call about that report. An agent handed this today would have to invent those answers.

Once they are settled — `/grill-with-docs` is the tool — this becomes `ready-for-agent` unchanged in scope.

**Comment by turner** [comment]
> *This was generated by AI during a `/grill-with-docs` session.*

Scope settled. Moving `ready-for-human` → `ready-for-agent`. Full reasoning, with the facts each decision was read off, is in **ADR-0010** (`docs/adr/0010-all-is-a-zoom-rung-not-a-chromosome.md`, PR #593); vocabulary in `CONTEXT.md`. Summary below.

## The reframe

Part (2) of the request cannot be done without part (1), and part (1) cannot be done by adding a number to a list. `minZoom` floors at the coarsest **declared** resolution and `setChromosomesView` clamps `pixelSize` up to `DEFAULT_PIXEL_SIZE`, so on a 2.4 Gbp scaffold the whole thing cannot be framed at all — the only coarser data in the file is the `All` matrix. So `All` disappears from the *vocabulary* while surviving as *data*: it becomes the sole scaffold's coarsest zoom rung. For a one-chromosome genome the cumulative offset is zero, so the transform is a unit match with no offset — jrobinso's 2018 warning about kb-from-genome-start coordinates degenerates to nothing in this case.

## Settled

**Predicate** — `HiCDataset.isSingleChromosome()`, true when `chromosomes.length === 2`, identifying `All` by `wholeGenomeChromosome` identity, not by name. The broader *"`All` and the sole chromosome describe the same view"* reading was **rejected**: "dominates" needs a threshold, and at 95%-one-scaffold the `All` view still shows the other 5%. If that is wanted it is a separate issue.

**Sentinel zoom index** — the synthetic rung carries a named `-1`, never a shifted real index, because `zoom` is a persisted wire value (`State.toJSON`). List sorted explicitly by descending `binSize`, which also fixes `interactionHandler`'s `resolutions[0]` / `resolutions[length-1]` direction guards.

**Bounded blast radius** — `browser.binSizeForZoom()` is introduced but only the sites the sentinel path touches are converted (ruler, `State` locus/bin math, tile source). The other ~15 `bpResolutions[state.zoom]` sites keep direct indexing; the accessor is a landing pad for #398.

**Two whole-genome tests diverge by design** — the tile source keeps `isWholeGenome` true at the sentinel (the ×4 threshold follows the *data*); `browser.isWholeGenome()` stays false (the vocabulary follows the *state*). #398 spotted this pair as possibly divergent; here it becomes load-bearing.

**Redirect at the state layer** — pulldown, locus-goto, session and URL all resolve `All` to the sole chromosome, silently, coercing rather than rejecting (ADR-0009). Saving from the sentinel writes `chr1: 0, chr2: 0, zoom: 0` — the same view, portable to every existing consumer, redirected back on restore. **No juicebox-web or Spacewalk coordination required.**

## Test plan

Synthetic single-chromosome dataset extending `test/utils/restoreDataset.js` — no `.hic` fixture needed, and none exists. Cover: pulldown omits `All`; restoring `chr1: 0` lands on the scaffold; the sentinel frames the whole scaffold; sentinel round-trips through save as the whole-genome view; and a normal multi-chromosome dataset is provably unchanged — that last is the regression that matters, since everything is gated on a predicate that must stay false for every real genome.

## Relationship to #398

**Separable, and deliberately so.** #398 is `needs-info` and blocked on a reporter who may not return; this should not wait on it. If #398's whole-genome hypothesis proves correct, this change removes that code path for exactly the assembly shape #398 reports.
